src/Models/NonhydrostaticModels/solve_for_pressure.jl

```diff
@@ -3,16 +3,26 @@ using Oceananigans.Solvers: FFTBasedPoissonSolver, FourierTridiagonalPoissonSolv
 using Oceananigans.Architectures: device_event
 using Oceananigans.Distributed: DistributedFFTBasedPoissonSolver
 
+using PencilArrays: Permutation
+
 #####
 ##### Calculate the right-hand-side of the non-hydrostatic pressure Poisson equation.
 #####
 
+const ZXYPermutation = Permutation{(3, 1, 2), 3}
+const ZYXPermutation = Permutation{(3, 2, 1), 3}
+
```

Comment:
These were also defined in `distributed_fft_based_poisson_solver.jl`. Is this double definition necessary or can we define this in only one place to reduce code?

Reply:
There's often a tradeoff between code reuse and code locality. Here I chose to "redefine" code so that this file could be understood without having to refer to `distributed_fft_based_poisson_solver.jl`. I reasoned that code reuse in this case held almost no advantage (`ZXYPermutation` is about as simple as it gets so there's little cost to repeating it), but that the gains in code locality were significant.

Reply:
There's no mathematical principle to deduce when code locality trumps reuse, or vice versa. I think it often makes to liberally redefine constants. For example, I could easily dispatch on `Permutation{(3, 1, 2), 3}` rather than `ZXYPermutation`. I just think that `ZXYPermutation` is more readable. On the other hand, if I import `ZXYPermutation` from elsewhere then I risk defeating the original purpose of making the code more readable, since the reader might have to find that other file.

But for anything but the most trivial functions, I think it usually makes sense to reuse rather than repeat.

Reply:
Agreed